features/netsocket/TCPServer.cpp

```diff
@@ -52,7 +53,8 @@ nsapi_error_t TCPServer::accept(TCPSocket *connection, SocketAddress *address)
             connection->_socket = socket;
             connection->_event = Callback<void()>(connection, &TCPSocket::event);
             _stack->socket_attach(socket, &Callback<void()>::thunk, &connection->_event);
-
+            _socket_stats.stats_update_peer(this, *address);
+            _socket_stats.stats_update_socket_state(this, SOCK_OPEN);
```

Comment:
Should these two lines actually be
```
            _socket_stats.stats_update_peer(connection, *address);
            _socket_stats.stats_update_socket_state(connection, SOCK_OPEN);
```
So it is not `this` but the new `connection` that we want to update the stats for.

Reply:
And after `accept()` the TCPSocket should be in `SOCK_CONNECTED` state. Not open.